### ISSUE TITLE
With 4.9.7, getting "do not know how to get a iprofile from the database" upon export

### ISSUE THREAD

**Issue by kirkpabk** [archi-contribs/database-plugin#171]
With a postgres database, running 4.9.7 plugin on version 5.0.2 of Archi, we get this when attempting to export.   Had 5 successful exports, this one failed.  It has been roughly a week since the last push.  The most major things that occurred are renaming of objects, views.  Relocation of objects/views within new or renamed folders.   But would assume that this sort of action would not cause this issue.

![2023-11-26_18-08-30](https://github.com/archi-contribs/database-plugin/assets/6099696/436aed18-3b51-4968-a0fa-e25a3ffdaa4e)

STACK TRACE
java.lang.Exception: Do not know how to get a IProfile from the database.
Do not know how to get a IProfile from the database.
  org.archicontribs.database.connection.DBDatabaseImportConnection.getObjectFromDatabase(DBDatabaseImportConnection.java:203)
  org.archicontribs.database.model.commands.DBImportProfileFromIdCommand.<init>(DBImportProfileFromIdCommand.java:72)
  org.archicontribs.database.GUI.DBGuiExportModel.export(DBGuiExportModel.java:2269)
  org.archicontribs.database.GUI.DBGuiExportModel$1.widgetSelected(DBGuiExportModel.java:133)
  org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
  org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
  org.eclipse.swt.widgets.Display.sendEvent(Display.java:4256)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
  org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4054)
  org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3642)
  org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
  org.eclipse.jface.window.Window.open(Window.java:799)
  com.archimatetool.editor.diagram.actions.FindReplaceAction.run(FindReplaceAction.java:34)
  org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
  org.eclipse.jface.commands.ActionHandler.execute(ActionHandler.java:121)
  org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
  jdk.internal.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
  java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
  java.base/java.lang.reflect.Method.invoke(Unknown Source)
  org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
  org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:317)
  org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:251)
  org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
  org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
  org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
  org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
  org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.executeCommand(KeyBindingDispatcher.java:308)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.press(KeyBindingDispatcher.java:580)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.processKeyEvent(KeyBindingDispatcher.java:647)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.filterKeySequenceBindings(KeyBindingDispatcher.java:439)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher$KeyDownFilter.handleEvent(KeyBindingDispatcher.java:96)
  org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
  org.eclipse.swt.widgets.Display.filterEvent(Display.java:1286)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1065)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1090)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1075)
  org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1117)
  org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1113)
  org.eclipse.swt.widgets.Widget.wmChar(Widget.java:1594)
  org.eclipse.swt.widgets.Control.WM_CHAR(Control.java:4865)
  org.eclipse.swt.widgets.Canvas.WM_CHAR(Canvas.java:345)
  org.eclipse.swt.widgets.Control.windowProc(Control.java:4743)
  org.eclipse.swt.widgets.Canvas.windowProc(Canvas.java:340)
  org.eclipse.swt.widgets.Display.windowProc(Display.java:5022)
  org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
  org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3640)
  org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
  org.eclipse.jface.window.Window.open(Window.java:799)
  com.archimatetool.editor.diagram.actions.FindReplaceAction.run(FindReplaceAction.java:34)
  org.eclipse.jface.action.Action.runWithEvent(Action.java:474)
  org.eclipse.jface.commands.ActionHandler.execute(ActionHandler.java:121)
  org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
  jdk.internal.reflect.GeneratedMethodAccessor50.invoke(Unknown Source)
  java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
  java.base/java.lang.reflect.Method.invoke(Unknown Source)
  org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
  org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:317)
  org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:251)
  org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
  org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
  org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
  org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:485)
  org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.executeCommand(KeyBindingDispatcher.java:308)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.press(KeyBindingDispatcher.java:580)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.processKeyEvent(KeyBindingDispatcher.java:647)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher.filterKeySequenceBindings(KeyBindingDispatcher.java:439)
  org.eclipse.e4.ui.bindings.keys.KeyBindingDispatcher$KeyDownFilter.handleEvent(KeyBindingDispatcher.java:96)
  org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
  org.eclipse.swt.widgets.Display.filterEvent(Display.java:1286)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1065)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1090)
  org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1075)
  org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1117)
  org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1113)
  org.eclipse.swt.widgets.Widget.wmChar(Widget.java:1594)
  org.eclipse.swt.widgets.Control.WM_CHAR(Control.java:4865)
  org.eclipse.swt.widgets.Canvas.WM_CHAR(Canvas.java:345)
  org.eclipse.swt.widgets.Control.windowProc(Control.java:4743)
  org.eclipse.swt.widgets.Canvas.windowProc(Canvas.java:340)
  org.eclipse.swt.widgets.Display.windowProc(Display.java:5022)
  org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
  org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3640)
  org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
  org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
  org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
  org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
  org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
  org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
  org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
  org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
  com.archimatetool.editor.Application.start(Application.java:70)
  org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
  org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
  org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
  org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
  org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
  java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
  java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
  java.base/java.lang.reflect.Method.invoke(Unknown Source)
  org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
  org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
  org.eclipse.equinox.launcher.Main.run(Main.java:1467)

**Comment by herve91** [comment]
It's a known issue. I'll send you a fix asap

**Comment by alf** [comment]
Has this been fixed? I get the same trying to export to a sqlite database.

**Comment by herve91** [comment]
Please check if the profiles and profiles_in_model tables do exist in your database. If not, you need to create them

**Here is SQL Server syntax:**
```
CREATE TABLE profiles (id VARCHAR(50) NOT NULL, version INT NOT NULL, name VARCHAR(1024), is_specialization TINYINT, image_path VARCHAR(50), concept_type VARCHAR(50), created_by VARCHAR(30) NOT NULL, created_on DATETIME NOT NULL, checkedin_by VARCHAR(30), checkedin_on DATETIME, deleted_by VARCHAR(30), deleted_on DATETIME, checksum VARCHAR(50) NOT NULL, PRIMARY KEY (id, version))

CREATE TABLE profiles_in_model (pim_id INT IDENTITY NOT NULL, profile_id VARCHAR(50) NOT NULL, profile_version INT NOT NULL, model_id VARCHAR(50) NOT NULL, model_version INT NOT NULL, pos INT NOT NULL, PRIMARY KEY (pim_id))
```

**Here is sqlite syntax:**
```
CREATE TABLE profiles (id VARCHAR(50) NOT NULL, version INTEGER(10) NOT NULL, name VARCHAR(1024), is_specialization TINYINT, image_path VARCHAR(50), concept_type VARCHAR(50), created_by VARCHAR(30) NOT NULL, created_on TIMESTAMP NOT NULL, checkedin_by VARCHAR(30), checkedin_on TIMESTAMP, deleted_by VARCHAR(30), deleted_on TIMESTAMP, checksum VARCHAR(50) NOT NULL, PRIMARY KEY (id, version))

CREATE TABLE profiles_in_model (pim_id INTEGER NOT NULL, profile_id VARCHAR(50) NOT NULL, profile_version INTEGER(10) NOT NULL, model_id VARCHAR(50) NOT NULL, model_version INTEGER(10) NOT NULL, pos INTEGER(10) NOT NULL, PRIMARY KEY (pim_id))
```

**Comment by herve91** [comment]
fixed in version 4.9.8 that will be released very soon
